registry/registry.go

```diff
@@ -1142,7 +1142,23 @@ func RemoveAppFromSpace(s *space.Space, appSlug string) error {
 		return err
 	}
 
-	app.Versions, err = FindAppVersionsCacheMiss(s, appSlug, Dev, Concatenated)
+	// First remove all pending versions for that app
+	pendingVersions, err := GetAppPengindVersions(s, appSlug)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range pendingVersions {
+		if err = DeletePendingVersion(s, v); err != nil {
+			fmt.Printf("Unable to delete pending version %s/%s: %s\n", v.Slug, v.Version, err)
```

Comment:
We should probably use a logger instead of printing on stdout

Reply:
yeah, this should have been done previously for the other `fmt.Printf()` in this file too, but it's out of scope for this PR.